entity-framework/core/querying/sql-queries.md

```diff
@@ -0,0 +1,163 @@
+
+While this syntax may look like regular C# [string interpolation](https://learn.microsoft.com/dotnet/csharp/language-reference/tokens/interpolated), the supplied value is wrapped in a `DbParameter` and the generated parameter name inserted where the `{0}` placeholder was specified. This makes > The <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql%2A> safe from SQL injection attacks, and sends the value efficiently and correctly to the database.
+
+When executing stored procedures, it can be useful to use named parameters in the SQL query string, especially when the stored procedure has optional parameters:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureNamedSqlParameter)]
+
+If you need more control over the database parameter being sent, you can also construct a `DbParameter` and supply it as a parameter value. This allows you to set the precise database type of the parameter, or facets such as its size, precision or length:
+
+[!code-csharp[Main](../../../samples/core/Querying/SqlQueries/Program.cs#FromSqlStoredProcedureSqlParameter)]
+
+> [!NOTE]
+>
+> Pay attention to the ordering of your parameters when executing a stored procedure. Unless using named parameters, the parameter in the SQL must exactly match the stored procedure definition, otherwise you may encounter type conversion exceptions and/or unexpected behavior when the procedure is executed.
+
+### Dynamic SQL and parameters
+
+<xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql%2A> and its parameterization should be used wherever possible. However, there are certain scenarios where SQL needs to be dynamically pieced together, and database parameters cannot be used. For example, let's assume that a C# variable holds the name of the a property to be filtered by. It may be tempting to use a SQL query such as the following:
+
+```c#
+var propertyName = "User";
+var propertyValue = "johndoe";
+
+var blogs = context.Blogs
+    .FromSql($"SELECT * FROM [Blogs] WHERE {propertyName} = {propertyValue}")
+    .ToList();
+```
+
+This code doesn't work, since database do not allow parameterizing column names (or any other part of the schema).
+
+First, carefully consider whether you really want to fully parameterize the column name. For example, users may choose a column that isn't indexed, making the query run extremely slowly and overload your database. Except for truly dynamic scenarios, it's usually better to have two SQL queries for two column names, rather than using parameterization to collapse them into a single SQL query.
+
+If you've decided you do want to use parameterization, you'll have to use <xref:Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSqlRaw%2A>, which allows interpolating variable data directly into the SQL string, instead of using a database parameter:
```

Comment:
I don't think this is true. Using `Where(e => EF.Property<object>(thePropertyName) == x)` is a better way of doing this.

Reply:
Oh sure, the idea was "if you've decided to use parameterization and have to do SQL as opposed to LINQ" (somewhat implied in the page context). I'll amend to make sure this is clearer.

Reply:
But people might think that the _only way to do this_ is with raw SQL.

Reply:
Tweaked the wording to make it clear that we're discussing dynamic construction only in the SQL context.